server/handlers/surveyHandler.js

```diff
@@ -0,0 +1,31 @@
+const Joi = require('joi');
+
+const Session = require('../models/Session');
+const MessageRepository = require('../repositories/MessageRepository');
+const model = require("../models/Message");
+
+
+
+
+const get = async (req, res, next) => {
+  await Joi.object({
+    uuid: Joi.string().required(),
+  }).unknown(false).validateAsync(req.params, { abortEarly: false });
+  const session = new Session();
+  const messageRepository = new MessageRepository(session);
+
+  try {
+    const result = await model.getSurveyReport(messageRepository, req.params.uuid);
+    res.status(200).send(result);
+  } catch (e) {
+    console.log(e);
+    if (session.isTransactionInProgress()) {
+      await session.rollbackTransaction();
+    }
+    res.status(422).json({ ...e });
+  }
+};
```

Comment:
I have standardized the message handler around the concept that repositories are only known to the model, and the session is only know to the service layer.  Could you review how I have the Messages stuff set up and conform this survey report implementation to that pattern?